File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int max1 = 0;
        int l = 0;
        int r = heights.size() -1;

        while(l < r){
            int w = r-l;
            int h = min(heights[l] , heights[r]);
            max1 = max(max1 , w *h);
            if(heights[l] < heights[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return max1;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxp = 0
        minbuy = prices[0]

        for n in prices:
            maxp = max(maxp,n-minbuy)
            minbuy = min(minbuy,n)
        return maxp

            